File: blog-frontend/blog-admin/src/services/resources/tag-service.ts
import { resourceAxios } from "../../utils/axios-utils"

export const tagService = {

    getAll: async () => {
        return await resourceAxios.get<void, Tag[]>('/tags/all')
    },

    getTags: async ({ pageSize = 5, pageNum = 0, name, description }: GetTags) => {
        return await resourceAxios.get<void, PageRespsone<Tag>>('/tags', {
            params: {
                pageSize,
                pageNum,
                ...(name !== '' ? { name } : {}),
                // ...(description ? { description } : {})
            }
        })
    },

    createTag: async ({ name, description }: CreateTag) => {
        return await resourceAxios.post<void, Tag>('/tags', {
            name,
            description
        })

    },

    updateTag: async (data: Tag) => {
        const { id, name, description } = data
        await resourceAxios.put<void, void>(`/tags/${id}`, {
            name: data.name,
            description: data.description
        })
    },

    deleteTag: async (id: number) => {
        await resourceAxios.delete<void, void>(`/tags/${id}`)
    }


}

File: blog-backend/blog-core/src/main/java/com/zach/blog/service/TagService.java
package com.zach.blog.service;

import com.zach.blog.model.ApplicationUser;
import com.zach.blog.model.Tag;
import org.springframework.data.domain.Page;

import java.util.List;

public interface TagService {

    Page<Tag> getTags(Integer pageNum, Integer pageSize, String tagName, String description);

    Tag createTag(Long userId, String name, String description);

    void deleteTag(Long userId, Long tagId);

    void updateTag(Long userId, Long tagId, String name, String description);

    List<Tag> getAllTags();
}
